sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileScanRDD.scala

```diff
@@ -70,6 +70,7 @@ class FileScanRDD(
     readFunction: (PartitionedFile) => Iterator[InternalRow],
     @transient val filePartitions: Seq[FilePartition],
     val readDataSchema: StructType,
+    val partitionSchema: StructType = new StructType(),
```

Comment:
Instead of passing two parameters, I think we just need a `readSchema: StrucType`. It's only used to create a projection to add metadata columns.

Reply:
Done.